### PR TITLE
Added Campaigns functionality.

### PR DESCRIPTION
**Pull request by ekoval**
https://github.com/yola/production/issues/2116
Also, test code is re-structured a bit to be more scalable.

Now tests have the following structure:
```
tests/base.py:SitewitTestCase
        |______tests/accounts/base.py:AccountTestCase(SitewitTestCase)
        |             |____tests for Accounts(AccountTestCase)
        |
        |______tests/campaigns/base.py:CampaignTestCase(SitewitTestCase)
                      |____tests for Campaigns(CampaignTestCase)
```

### CODE REVIEW

File: tests/campaigns/test_integration.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+from mock import Mock, patch
```
Comment (snitch) [inline tests/campaigns/test_integration.py]: F401 'Mock' imported but unused

File: tests/campaigns/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        self.result = service.list_campaigns(self.account_token)
+
+    def test_campaigns_list_is_returned(self):
+        self.assertEqual(sorted(self.result), sorted(self.campaigns))
+
+
+class TestListCampaignsBadAccountToken(CampaignTestCase):
+
+    def setUp(self):
+        service = SitewitService()
```
Comment (snitch) [inline tests/campaigns/test_integration.py]: F841 local variable 'service' is assigned to but never used

File: sitewit/services.py [hunk 3]
```diff
@@ -129,3 +129,43 @@ def generate_sso_token(self, user_token, account_token):
...
+        """Delete Campaign by campaign ID
+
+        Args:
+            campaign_id (str): id of campaign.
+
+        Returns:
+            dict of format:   {'id': 1, 'name': 'test', 'status': 'Active'}
+        """
+        return self.delete(
+            'api/Campaign/%s' % (campaign_id,),
```
Comment (RayeN) [inline sitewit/services.py]: lower case `campaign` in URL

File: tests/accounts/base.py [hunk 4]
```diff
@@ -1,15 +1,9 @@
...
-from unittest2 import TestCase
-from yoconfig import configure
-from yoconfigurator.base import read_config
+from tests.base import SitewitTestCase
 
 
-class BaseTestCase(TestCase):
+class AccountTestCase(SitewitTestCase):
     account_id = 9999
     site_id = str(uuid.uuid4())
```
Comment (RayeN) [inline tests/accounts/base.py]: real site_id is `uuid.uuid4().hex`

File: tests/campaigns/base.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+from tests.base import SitewitTestCase
+
+
+class CampaignTestCase(SitewitTestCase):
+    # We have to hardcode these values for integration testing, because SiteWit
+    # doesn't have Create Campaign API endpoint. Campaigns are created
```
Comment (RayeN) [inline tests/campaigns/base.py]: Looks like comment is unfinished.

File: tests/campaigns/test_integration.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+class TestListCampaignsBadAccountToken(CampaignTestCase):
+
+    def test_error_401_is_raised(self):
+        self.assertHTTPErrorIsRaised(
+            SitewitService().list_campaigns, (self.random_token,),
+            401, {u'Message': u'Malformed SubPartner Identifier'})
+
+
+class TestDeleteCampaign(CampaignTestCase):
```
Comment (RayeN) [inline tests/campaigns/test_integration.py]: probably we shouldn't call this test integration, then...

**Comment by RayeN** [comment]
:+1: Looks good to me.

**Pull request merged**